larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -538,7 +556,27 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
             }
 
             for (const ParticleFlowObject *const pPfoToShift : pfoVector)
+            {
+                CaloHitList caloHitList3D;
+                LArPfoHelper::GetCaloHits(pPfoToShift, TPC_3D, caloHitList3D);
+
+                float xsum(0.f);
+                int count(0);
+
+                for (const CaloHit *const pCaloHit : caloHitList3D)
+                {
+                    xsum += pCaloHit->GetPositionVector().GetX();
+                    ++count;
+                }
+
+                if (count == 0)
+                    continue;
+
+                const float averageX(xsum / static_cast<float>(count));
+                const float sign(averageX < tpcBoundaryCenterX ? 1.f : -1.f);
+                x0 = std::fabs(x0) * sign;
```

Comment:
Yes, this seems like a neat way to do it. I think this works for all cases now, but if you can reply to say you've checked it through one more time, that'd be great!